nemo/src/model/rule_model/aggregate.rs

```diff
@@ -7,6 +7,25 @@ pub struct Aggregate {
     pub(crate) variable_identifiers: Vec<Identifier>,
 }
 
+impl Aggregate {
+    /// Substitutes all occurrences of `var` for `subst`.
+    pub fn substitute_variable(&mut self, var: &Variable, subst: &Variable) {
+        let Variable::Universal(var) = var else {
+            panic!("cannot substitute existential variable")
+        };
+
+        let Variable::Universal(subst) = subst else {
+            panic!("cannot substitute existential variable")
```

Comment:
maybe be slightly more explicit and have `cannot substitute with existential variable` here?